infra.bs

```diff
@@ -941,6 +941,69 @@ ordering will not match any particular alphabet or lexicographic order, particul
 
 <hr>
 
+<p>The <dfn export>code unit substring</dfn> from index <var>start</var> with length
+<var>length</var> within a <a>string</a> <var>string</var> is determined as follows:
+
+<ol>
+ <li><p><a>Assert</a>: <var>start</var> and <var>length</var> are greater than 0.</p></li>
+
+ <li><p><a>Assert</a>: <var>start</var> + <var>length</var> is less than or equal to
+ <var>string</var>'s <a for=string>length</a>.</p></li>
```

Comment:
I wonder if this is the most useful interface for all callers as it means they have to check the length of each string as well and do a bunch of branching around it.

Reply:
Yeah, I saw that URL seems to want to go to the end of the string. We could make length optional in a followup.